core/src/main/scala/org/bitcoins/core/protocol/ln/currency/MilliSatoshis.scala

```diff
@@ -62,27 +63,63 @@ sealed abstract class MilliSatoshis extends NetworkElement {
   }
 
   def ==(ms: MilliSatoshis): Boolean = {
-    toLnCurrencyUnit == ms.toLnCurrencyUnit
+    toBigInt == ms.toBigInt
```

Comment:
Errors were appearing because there's `require` statements in `PicoBitcoin` etc. IMO there's no real reason to do these comparisons via `toLnCurrencyUnit`, as it's all `BigInt`s all the way down anyways